### PR TITLE
Remember all options after restart and add button to reset in settings

### LINKED ISSUE #57: Remember all options after restart
Please add the ability to remember hold or normal settings every time XClicker reopens

### PR DESCRIPTION
**Pull request by koutoftimer**
Closes #57 
I have added 2 buttons to save and load options to `~/.config/xclicker.conf`. It can be extended with file dialog to choose file location.

It is nowhere near perfect solution but it works for me.

Related to #57

### CODE REVIEW

**Comment by robiot** [comment]
Cool, will take a look asap

**Comment by koutoftimer** [comment]
@robiot I have decided to spend more time to figure out better solution. Right now I'm reusing `GKeyFile` from the `settings.c`.

At this stage, maybe it is good idea to save all options by default as in settings window and just add "delete/reset settings" button to it. But my initial idea with presets is that there may be multiple of them you can switch between them with ease. Well, if feels like a lot of work to implement both scenarios.

Sorry for not marking pull request as a draft, I was not about to make this change yesterday.

**Comment by robiot** [comment]
From what I have seen it looks like you did some amazing work. Tell me whenever you think its ready for review :)

**Comment by koutoftimer** [comment]
@robiot well, I guess you can review it now. I do not feel like adding more changes.

File: src/resources/ui/xclicker-window.ui [hunk 1]
```diff
@@ -238,7 +238,7 @@
                         <property name="visible">True</property>
                         <property name="can-focus">False</property>
                         <property name="label" translatable="yes">Mouse Button</property>
-                        <property name="xalign">0.2</property>
+                        <property name="xalign">0.20000000298023224</property>
```
Comment (robiot) [inline src/resources/ui/xclicker-window.ui:241]: very specific number, change back to 0.2

**Comment by robiot** [comment]
Cool, I will take a look.

File: src/resources/ui/xclicker-window.ui [hunk 2]
```diff
@@ -272,13 +272,12 @@
...
                       </packing>
                     </child>
-
                     <child>
                       <object class="GtkLabel">
                         <property name="visible">True</property>
                         <property name="can-focus">False</property>
                         <property name="label" translatable="yes">Click Type</property>
-                        <property name="xalign">0.08</property>
+                        <property name="xalign">0.07999999821186065</property>
```
Comment (robiot) [inline src/resources/ui/xclicker-window.ui:280]: very specific number, change back to 0.08

File: src/resources/ui/xclicker-window.ui [hunk 3]
```diff
@@ -558,14 +553,16 @@
...
-                        <signal name="insert-text" handler="insert_handler" swapped="no" />
+                        <signal name="insert-text" handler="insert_handler" swapped="no"/>
                       </object>
                       <packing>
                         <property name="left-attach">1</property>
                         <property name="top-attach">3</property>
                       </packing>
                     </child>
-
+                    <child>
```
Comment (robiot) [inline src/resources/ui/xclicker-window.ui:563]: Whats this?

**Review by robiot — CHANGES_REQUESTED** [review]
Looks good for the most part. I'm just going to test it as soon as possible, then I could merge.

File: src/resources/ui/xclicker-window.ui (hunk 3 above)
Comment (koutoftimer) [reply src/resources/ui/xclicker-window.ui:563]: I haven't touched ui file by hands. Glade does this..

Comment (koutoftimer) [reply src/resources/ui/xclicker-window.ui:563]: I guess, it will be the same for everyone who edit ui files in Glade.

**Comment by koutoftimer** [comment]
@robiot 

> I'm just going to test it as soon as possible

Do you have some ideas about setting up unittests?

**Comment by robiot** [comment]
Hm I couldn't get the load and save to work.
I also think the most optimal would be to have a just a save main window settings button in the Settings window, and then autoload.
Also maybe have a reset main window settings button there.

If you don't feel like fixing that, I could do it whenever I have time, which means it will take some time for me to get to it.

**Comment by koutoftimer** [comment]
@robiot my bad, I have broken it by trying to tidy up the code.

````
(xclicker:97278): Gtk-WARNING **: 06:54:50.253: Could not find signal handler 'load_preset_clicked'.  Did you compile with -rdynamic?

(xclicker:97278): Gtk-WARNING **: 06:54:50.253: Could not find signal handler 'save_preset_clicked'.  Did you compile with -rdynamic?
````

I have added those handers back.

**Comment by koutoftimer** [comment]
UPD: having an ability to select which preset to use is a one way to go. Automatic saving with no user interference is second. In order to satisfy both use cases I would have to implement both and UI should change according to which way user prefers. That is why I told that it is too much work for me. I have implemented the way that fits me best. We can discuss what and how to improve in the new issue, IMHO.

**Comment by robiot** [comment]
Closes #57

**Pull request merged**